src/common/src/DocumentViewer/DocumentViewer.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react';
+import { Alert, Heading, Spinner, themeColor } from '@amsterdam/asc-ui';
+import { DocumentViewerStyle } from './DocumentViewerStyles';
+import DocumentRenderer from './DocumentRenderer';
+
+type Props = {
+	uri: string;
+	authorizationHeader?: string;
+};
+
+export type DocumentState = {
+	uri: string;
+	loading: boolean;
+	filename?: string;
+	error?: string;
+	contentType?: string;
+};
+
+const DocumentViewer: React.FC<Props> = ({ uri, authorizationHeader }: Props) => {
+	const [documentState, setDocumentState] = useState<DocumentState>({ uri, loading: false });
+
+	const updateDocumentState = (documentState: Partial<DocumentState>) => {
+		setDocumentState((prevState) => {
+			return {
+				...prevState,
+				...documentState,
+			};
+		});
+	};
+
+	const getRequestHeaders = (): HeadersInit => {
+		if (authorizationHeader) {
+			return { Authorization: authorizationHeader };
+		}
+
+		return {};
+	};
+
+	useEffect(() => {
+		const filename = uri.split('/').pop() || 'Onbekend bestand';
+		updateDocumentState({ loading: true, error: undefined, contentType: undefined, filename });
+
+		fetch(uri, { method: 'HEAD', headers: getRequestHeaders() })
+			.then((response) => {
+				if (!response.ok) {
+					throw response.status === 404 ? 'Document niet gevonden.' : 'Fout bij het ophalen.';
```

Comment:
In this instance I think it's ok to throw a string, but keep in mind that if a 3rd party is catching the exception they will probably expect a full-blown error object.

Reply:
Changed it to error handling in the state and let other errors be thrown.